students/jjakub/session05/mailroom03.py

```diff
@@ -32,43 +46,54 @@ def send_thanks():
             amt = int(input("Please enter the donation amount: "))
             donor_db.setdefault(name,[]).append(amt)
             print(f'\nDear {name.capitalize()},\
-            \nThank you for the donation of ${amt:.2f}.\nSincerely,\
-            \nThe Mailroom Foundation')
+                    \nThank you for the donation of ${amt:.2f}.\nSincerely,\
+                    \nThe Mailroom Foundation')
             return
         except ValueError:
-            print("\n Warning: Please enter an integer!\n")
+            print("\n Warning: Please enter an integer! \n")
+
+
+def donor_calc(donor_db):
+    donor_sum = [
+            (donor.capitalize(), sum(amt), len(amt), sum(amt)/len(amt)) 
+            for donor, amt in donor_db.items()
+            ]
```

Comment:
Definitely a good list comprehension, though unfortunately `sum(amt)` is computed twice (unnecessarily repeated)... Let's focus more on the pedagogical aspect and call this fine, though in production, you wouldn't want to repeat the same computation twice unnecessarily. I'm not sure if this can be automatically optimized by Python runtime, and my guess is that it is not...